routers/posts.py

```
@@ -0,0 +1,79 @@
+    if not(post["author_id"] == int(user["id"])):
+        raise HTTPException(status_code=403, detail="User doesn't have permission")
+    if payload.title is not None:
+        post["title"] = payload.title
+    if payload.content is not None:
+        post["content"] = payload.content
+    write_json("posts.json",posts)
+    return post
+
+# 게시글 삭제
+@router.delete("/{post_id}")
+def delete_post(post_id: int, user=Depends(get_current_user)):
```

Ruff (0.14.13)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)